pkg/refcount/counter.go

```diff
@@ -0,0 +1,125 @@
+package refcount
```

Comment:
I find this package hard to use. It mixing storage + counter. `SetIncr` and `GetIncr` looks pretty weird.